src/ripple/ledger/impl/ReadView.cpp

```diff
@@ -165,11 +163,9 @@ ReadView::txs_type::begin() const -> iterator
 }
 
 auto
-ReadView::txs_type::end() const -> iterator const&
+ReadView::txs_type::end() const -> iterator
 {
-    if (!end_)
-        end_ = iterator(view_, view_->txsEnd());
-    return *end_;
+    return iterator(view_, view_->txsEnd());
```

Comment:
why not keep the caching/lazy evaluation and just use a mutex to avoid the race condition?